commands/uploader.go

```diff
@@ -24,10 +24,17 @@ type uploadContext struct {
 	committerName  string
 	committerEmail string
 
-	locks          map[string]locking.Lock
 	trackedLocksMu *sync.Mutex
-	ownedLocks     []locking.Lock
-	unownedLocks   []locking.Lock
+
+	// ALL verifiable locks
+	ourLocks   map[string]locking.Lock
+	theirLocks map[string]locking.Lock
+
+	// locks from ourLocks that were modified in this push
+	ownedLocks []locking.Lock
```

Comment:
I'd rather keep it simple. I'm not sure users will be creating enough locks to matter.